src/components/line-chart/line-chart.component.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { LineChartComponent } from './line-chart.component';
+import { DataModule } from '../../services/data/data.module';
+import { SimpleChange } from '@angular/core';
+import { mockDatum } from '../../utils/mocks.spec';
+
+describe('LineChartComponent', () => {
+  let fixture: ComponentFixture<LineChartComponent>;
+  let component: LineChartComponent;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        DataModule,
+      ],
+      declarations: [
+        LineChartComponent,
+      ],
+    });
+    fixture = TestBed.createComponent(LineChartComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should instantiate.', () => {
+    expect(component).toBeInstanceOf(LineChartComponent);
+  });
+
+  it('should have truthy i18n values.', () => {
+    component.activeDatum = mockDatum;
+    expect(component.ACTIVE_DATUM).toBeTruthy();
+  });
+
+  it('should synchronize data with data$.', () => {
+    component.data = [];
```

Comment:
We have been trying to write new tests using the Arrange - Act - Assert structure. So all the set up work is done in one block at the start of the test, followed by some actions, followed by assertions. It has helped us to keep our tests a bit cleaner. What do you think about using that pattern here?

Reply:
I see the pattern can make some tests cleaner, but wouldn't it be less straightforward for the tests where the order of `expect` statements matters? If I were to follow the pattern in this test, it adds unnecessary lines to store the value of the variables after each 'Act'.

```ts
it('should synchronize data with data$.', () => {
  component.data = [];
  const observedValue1 = component.data$.value;
  const value1 = component.data;

  component.data = [mockDatum];
  const observedValue2 = component.data$.value;
  const value2 = component.data;

  expect(observedValue1).toBe(value1);
  expect(observedValue2).toBe(value2);
});
```

Reply:
Do you need to test `observedValue2`? It seems to me that testing one instance of setting data should be sufficient. Or if not, you could break the test into two. I.e.`it('should set data as data$`, ...)` and 'it('should update data when data$ changes', ...)`.